### ISSUE TITLE
Integrate automatic A11y testing with AXE-Core on component level

### ISSUE THREAD

**Issue by o-hook** [porsche-design-system/porsche-design-system#1919]
Scope
---
To shift-left accessibility testing to fetch possible issues earlier in the development phase, we have to create automatic a11y testings for all of our design system components and integrate it into our pipeline. The open source software AXE-Core and AXE-Core-NPM as testing framework should be used.

Out of Scope
---
- States VRT Pages shall not be scanned by Axe Core

Notes  
---
- Use `components-js` VRT pages for a11y testing
  - What about iFrame test pages like it's the case for Flyout, Flyout Navigation, Modal and Toast?
  - What about components that have hidden UI elements (option list) like Select-Wrapper or Multi Select?
- Create a new Playwright a11y test folder with specs (e.g. `some-test-file.a11y.ts`) and test setup, like it's already done for `tests/{unit|e2e|vrt}` => `tests/{unit|e2e|a11y|vrt}`
- What happens to already existing a11y snapshots and a11y e2e test cases?
  - move to new `test/a11y` folder
  - a11y snapshot test might be removed at all (at least Playwright does not recommend them as soon as Axe Core is implemented)
  - In the future (not with this ticket), extend and focus more on custom keyboard flow tests and custom aria tests depending on certain interaction with mouse or keyboard
- By default only test on viewport width `1000px` (desktop) and `320px` (mobile)
- Fix possible issues (or create follow up issues in case it's to much effort)
- Find a solution how to deal with found accessibility issues and how strict deployment should be (e.g. in which case should the deployment be stopped in regards of the severity of issues - critical, serious, moderate and minor)
- AXE-Core Testing Engine / Rulesets: https://github.com/dequelabs/axe-core
- AXE-Core packages: https://github.com/dequelabs/axe-core-npm
- AXE-Core API docs: https://github.com/dequelabs/axe-core/blob/master/doc/API.md
- https://playwright.dev/docs/accessibility-testing
- https://playwright.dev/docs/api/class-accessibility#accessibility-snapshot

Acceptance criteria
---
- [x] Automated accessibility testing with Axe Core and conformance level of WCAG 2.2 AA for all design system components is integrated

Subtasks
---
- [x] WCAG 2.2 AA `target-size` rule is not excluded anymore
